### ISSUE TITLE
invalid_grant error message

### ISSUE THREAD

**Issue by osca2000** [IdentityServer/IdentityServer4#334]
I set up "identityserver 4 dev" in win 2012 server with valid SSL certificate. 
(Initially I tried to make the server (http://localhost:1941) with a sample mobile app but it didn't work without valid SSL certificate. )
Finally the sample oidc UWP app can access my test IDsvr 4.
However, the last step in authentication fails with an error message 'invalid_grant'.
1. Click Login and login screen appears in the mobile emulator.
2. Login successfully with user name = bob and password = bob.
3. Permission page is shown. 
4. When I click 'Yes Allow' button, I get the error message 'Invalid_grant'. Anyone knows how to fix this error?

Thanks

**Comment by leastprivilege** [comment]
What do the logs say?

**Comment by osca2000** [comment]
You mean IIS logs or IDsvr4 logs? I enabled all logs in startup.cs file but I cannot find any specific log files or log entries in Windows related to your seriLog. I may be wrong but I cannot run in debug mode IdSvr4 for mobile app because of SSL certificate issue.
Anyway when I checked IIS log, it was slightly different from one (successful) other (failed) but it is hard to find good debug information. 

Anyway I played with the client and scope and finally it works for me now. There may be better way but I don't know. 
For others, I like to list down a bit what I did. MVC or web works with IdSvr4 straightway but not mobile app.
1. IIS express temp SSL doesn't work because the mobile emulator thinks it is not part of its hosting pc. Valid SSL makes the mobile go further.
2. The RedirectURI in both mobile and IdSvr4 client must be identical.
3. Three entries seem to be required in the Client.
AccessTokenType = AccessTokenType.Jwt,
RequirePkce=true,
AllowedGrantTypes = GrantTypes.Hybrid,

Anyway, is it possible to test mobile apps without valid SSL certificate?

Thanks
